### ISSUE TITLE
Symbols of excluded files aren't listed in goto symbols

### ISSUE THREAD

**Issue by MichaReiser** [astral-sh/ty#3418]
> I'm playing with ty and Zed and it appears that if I set exclude = ["tests"] for ty, I lose code intelligence like cmd-t but I emphatically do not want any type-checking for my tests. Is there some trick so include tests for the LSP but disable all typing errors without enumerating them all?
> Originally posted on [Discord](https://discord.com/channels/1039017663004942429/1501542012774645932)

This can be worked around by using 

```toml
[tool.ty]

[[tool.ty.overrides]]
include = ["base.py"]
rules.all = "ignore"
```

But this isn't very intuitive and also results in ty type checking all test files. It just doesn't emit any diagnostics. The current approach does have the benefit over `exclude` that syntax errors are still reported.
